### ISSUE TITLE
Image Memory Leak? Help Needed.

### ISSUE THREAD

**Issue by PPeanutButter** [JetBrains/compose-jb#2691]
```kotlin
@Composable
fun CoverImage(albumRelativePath: String, modifier: Modifier = Modifier) {
    var imageBitmap by remember { mutableStateOf<ImageBitmap?>(null) }
    imageBitmap?.let {
        Image(bitmap = it, contentDescription = null, modifier = modifier.aspectRatio(2f/3f), contentScale = ContentScale.Crop)
    }
    LaunchedEffect(Unit) {
        MainScope().launch(Dispatchers.IO) {
//println(imageBitmap==null)
            val api = ApiService.getInstance(Configuration.Empty)
            val r = api.getCover(albumRelativePath)
            r.byteStream().let {
                var a = ImageIO.read(it)
                a = ImageResizer.resize(a, 300, 450)
                imageBitmap = a.toBitmap().asComposeImageBitmap()
            }
        }
    }
}
...
LazyVerticalStaggeredGrid(
        columns = StaggeredGridCells.Fixed(2),
        modifier = Modifier.padding(start = 4.dp, end = 4.dp)) {
        items(items = episodes, key = { item: Episode -> item.hashKey }){
            CoverImage(...)
        }
    }

if(show){
                EpisodeScreen(...){LazyVerticalStaggeredGrid(...)}
            }
```
I use this to display a url image in windows, after show/off `EpisodeScreen`, mem uasge gets larger。
mem uasge should be the same?(off vs show->off)
and every time make show=true, when LaunchedEffect(Unit) execed, `//println(imageBitmap==null)` awayls null(seems like there is no mem cache)
help wanted pls...

**Comment by eymar** [comment]
> and every time make show=true, when LaunchedEffect(Unit) execed, //println(imageBitmap==null) awayls null(seems like there is no mem cache)

That's true and it's expected. Did you expect it to behave differently? Or you made an assumption that the mem usage growth could be caused by some ImageBitmap caching?
____

> mem uasge gets larger。mem uasge should be the same?(off vs show->off)
It depends on the Garbage Collection. It could've not cleaned yet.

You may try to call `System.gc()` when show switches from true to false and see if it makes any difference. Let me know if it gives more info.

___
From your example I'm not sure I understand who is your suspect for the potential memory leak. 
At first I thought you're blaming the `CoverImage`, but

I see `CoverImage ` being used here:
```kotlin
items(items = episodes, key = { item: Episode -> item.hashKey }){
            CoverImage(...)
        }
```
where it's independent of `show` value.

**Comment by PPeanutButter** [comment]
I suspect for the potential memory leak is imageBitmap, app start with 200MB, stable with ~3GB in `EpisodeScreen`, then go back it may use 400MB(thought gc called by compose, but some imageBitmap not been released), a few more cycle, may stable with 1.2GB.
After call `System.gc()` seems worked! All `imageBitmap` are released.
Thanks for your analysis and answers

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2691">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
